prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -1316,12 +1376,23 @@ ssize_t rxr_pkt_init_long_rtw(struct rxr_ep *ep,
 	assert(tx_entry->op == ofi_op_write);
 
 	rtw_hdr = (struct rxr_long_rtw_hdr *)pkt_entry->pkt;
-	rtw_hdr->rma_iov_count = tx_entry->rma_iov_count;
-	rtw_hdr->data_len = tx_entry->total_len;
-	rtw_hdr->tx_id = tx_entry->tx_id;
-	rtw_hdr->credit_request = tx_entry->credit_request;
-	rxr_pkt_init_req_hdr(ep, tx_entry, RXR_LONG_RTW_PKT, pkt_entry);
+	rxr_pkt_init_long_rtw_hdr(ep, tx_entry, pkt_entry, RXR_LONG_RTW_PKT);
+	rxr_pkt_init_rtw_data(ep, tx_entry, pkt_entry, rtw_hdr->rma_iov);
+	return 0;
+}
+
+ssize_t rxr_pkt_init_dc_long_rtw(struct rxr_ep *ep,
+				 struct rxr_tx_entry *tx_entry,
+				 struct rxr_pkt_entry *pkt_entry)
+{
+	struct rxr_long_rtw_hdr *rtw_hdr;
+
+	assert(tx_entry->op == ofi_op_write);
+
+	rtw_hdr = (struct rxr_long_rtw_hdr *)pkt_entry->pkt;
+	rxr_pkt_init_long_rtw_hdr(ep, tx_entry, pkt_entry, RXR_DC_LONG_RTW_PKT);
 	rxr_pkt_init_rtw_data(ep, tx_entry, pkt_entry, rtw_hdr->rma_iov);
+
```

Comment:
Maybe I'm missing something, but some functions are pretty much the same for DC and non-DC versions. For example, comparing `rxr_pkt_init_long_rtw` and `rxr_pkt_init_dc_long_rtw`, the only difference is pkt_type. Is there any way to remove these duplicated code?

Reply:
Currently `rxr_pkt_init_long_rtw` and `rxr_pkt_init_dc_long_rtw` reduce code duplication via both calling `rxr_pkt_init_long_rtw_hdr` and `rxr_pkt_init_rtw_data`. 

But I think keeping `rxr_pkt_init_long_rtw ` and `rxr_pkt_init_dc_long_rtw` will make the code much more clearer. In `rxr_pkt_init_ctrl`, given different packet type, different init function is called respectively.